client/signup/steps/domains/index.jsx

```diff
@@ -61,6 +61,25 @@ const DomainsStep = React.createClass( {
 		return { products: productsList.get() };
 	},
 
+	componentWillMount: function() {
+		const { stepName, queryObject } = this.props;
+
+		// skip this step if domain name is given in the url via the `domain` query parameter
+		if ( queryObject && queryObject.domain ) {
```

Comment:
I think its simpler to just remove the domain step altogether. If users do want to edit their domain they can just navigate back to /domains which will load fast.